src/main/java/nextstep/session/service/CoverServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package nextstep.session.service;
+
+import nextstep.common.domain.BaseEntity;
+import nextstep.common.domain.DeleteHistory;
+import nextstep.exception.CoverException;
+import nextstep.session.domain.Cover;
+import nextstep.session.domain.ImageFilePath;
+import nextstep.session.domain.Resolution;
+import nextstep.session.dto.CoverDto;
+import nextstep.session.infrastructure.CoverRepository;
+import nextstep.users.domain.NsUser;
+import nextstep.users.infrastructure.UserService;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Resource;
+import java.time.LocalDateTime;
+
+@Service("coverService")
+public class CoverServiceImpl implements CoverService {
+
+    public static final int NO_UPDATE_COUNT = 0;
+    @Resource(name = "coverRepository")
+    private CoverRepository coverRepository;
+
+    @Resource(name = "userService")
+    private UserService userService;
+
+    @Override
+    public Cover findById(Long coverId) {
+        CoverDto coverDto = coverRepository.findById(coverId);
+        NsUser nsUser = userService.findByUserId(coverDto.getWriterId());
```

Comment:
서비스가 다른 서비스를 의존하게 개발하셨군요!
이부분에 대해 여러 의견들이 있어 관련 링크 남겨드릴게요.

레이어간 참조 가능
https://www.inflearn.com/questions/987690/%EC%84%9C%EB%B9%84%EC%8A%A4-%EB%A0%88%EC%9D%B4%EC%96%B4-%EB%81%BC%EB%A6%AC%EC%9D%98-%EC%9D%98%EC%A1%B4

지양하는 글
https://geminikims.medium.com/%EC%A7%80%EC%86%8D-%EC%84%B1%EC%9E%A5-%EA%B0%80%EB%8A%A5%ED%95%9C-%EC%86%8C%ED%94%84%ED%8A%B8%EC%9B%A8%EC%96%B4%EB%A5%BC-%EB%A7%8C%EB%93%A4%EC%96%B4%EA%B0%80%EB%8A%94-%EB%B0%A9%EB%B2%95-97844c5dab63

개인적으론 서비스가 다른 서비스를 의존하는 행위를 지양하는 편이긴 합니다.
요구사항이 고도화됨에 따라 의존성 사이클이 걸릴 가능성이 높기 때문이에요 🤔

현재 구조에서 cover 정보를 조회한 뒤 유저의 대한 정보도 추가로 필요해보여 의존한것으로 보입니다.

```suggestion
        NsUser nsUser = userRepository.findByUserId(coverDto.getWriterId());
```

위와 같이 변경되어도 문제없는거 아닐까요?

Reply:
이 부분도 오늘 강의를 들으며 고찰 해 보았습니다.

다만 생각이 든 것이 `userRepoistory`의 `findByUserId`는 DB에서 값을 `Optional`로 읽고 오고 있습니다.
그리고 userService`의 `findByUserId`는 레포지토리에서 받은 `Optional`을 검증하는 역할까지 추가로 해주고 있습니다.

`userRepository`의 기능을 그대로 쓰면, 여기저기 검증 로직이나 Service레이어의 다른 로직들을 다른 도메인에서 다시 구현하게 되고, 결국 코드의 중복이 발생될 텐데 "A Service에서 B  Service가 아닌 B Repoistory"를 참조하는게 맞을까?"
란 고민을 하게 되었습니다.

리뷰어님의 생각이 어떠실지 궁금합니다!

Reply:
> userRepository의 기능을 그대로 쓰면, 여기저기 검증 로직이나 Service레이어의 다른 로직들을 다른 도메인에서 다시 구현하게 되고, 결국 코드의 중복이 발생될 텐데 "A Service에서 B Service가 아닌 B Repoistory"를 참조하는게 맞을까?"
란 고민을 하게 되었습니다.

현재 코드의 중복이 발생되지 않으니 패키지간의 참조가 일어나지 않는게 좋은 방향 아닐까요? 🤔
훗날 참조가 된다면 위와 같은 생각으로 바뀔 수 있다는 생각은 저도 동의합니다!
다만, 예외를 각 호출한 곳에서 핸들링해야한다면 결국 Repository 로 호출하는게 간편하기도 하기에 어떤게 득과 실이 많을지 충분한 고민이 필요해보여요 🤔